lib/spack/spack/build_environment.py

```diff
@@ -109,7 +109,7 @@ def __call__(self, *args, **kwargs):
         disable = env_flag(SPACK_NO_PARALLEL_MAKE)
         parallel = not disable and kwargs.get('parallel', self.jobs > 1)
 
-        if parallel:
+        if parallel and 'bazel' not in self.command:
```

Comment:
I think you can achieve the same without centralizing the knowledge that `bazel` exists. In the `bazel` package you can override `setup_dependent_module` and : 
- add `bazel` command from there
- set `parallel = False` to any dependee